File: SteamLanguageParser/src/main/java/in/dragonbra/steamlanguageparser/generator/JavaFileWriter.java
package in.dragonbra.steamlanguageparser.generator;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

/**
 * @author lngtr
 * @since 2018-02-16
 */
public class JavaFileWriter extends FileWriter {

    private static final String INDENTATION = "    ";

    private String indent = "";

    public JavaFileWriter(File file) throws IOException {
        super(file);
    }

    public void indent() {
        indent += INDENTATION;
    }

    public void unindent() {
        indent = indent.substring(INDENTATION.length());
    }

    public void writeln(String line) throws IOException {
        write(indent);
        write(line);
        writeln();
    }

    public void writeln() throws IOException {
        write('\n');
    }
}

File: buildSrc/src/main/kotlin/in/dragonbra/generators/steamlanguage/generator/JavaFileWriter.kt
package `in`.dragonbra.generators.steamlanguage.generator

import java.io.File
import java.io.FileWriter
import java.io.IOException

class JavaFileWriter(file: File) : FileWriter(file) {

    companion object {
        private const val INDENTATION = "    "
    }

    private var indent: String = ""

    fun indent() {
        indent += INDENTATION
    }

    fun unindent() {
        indent = indent.substring(INDENTATION.length)
    }

    @Throws(IOException::class)
    fun writeln(string: String) {
        write(indent)
        write(string)
        writeln()
    }

    @Throws(IOException::class)
    fun writeln() {
        write("\n")
    }
}
